src/lib/tls/asio/asio_stream.h

```diff
@@ -533,28 +566,32 @@ class Stream
       class StreamCore : public Botan::TLS::Callbacks
          {
          public:
-            StreamCore(boost::beast::flat_buffer& receive_buffer, boost::beast::flat_buffer& send_buffer, Context& context)
-               : m_receive_buffer(receive_buffer), m_send_buffer(send_buffer), m_tls_context(context) {}
+            StreamCore(Stream& stream) : m_stream(stream) {}
```

Comment:
I'm not really happy with using the stream reference here directly, and also not with accessing the private members of the stream directly below. This was changed because we need to communicate receiving `close_notify` to the stream in some way. We could keep the parameters as they were and add an additional `std::function<> alert_callback` or similar, but I'm not sure if it's worth it.